utils/db-parts.stanza

```diff
@@ -296,18 +319,33 @@ public defn Capacitor (raw-json: JObject) -> Capacitor :
   val json = filter-json(raw-json)
   val [x, y, z] = parse-dimensions(json["dimensions"] as JObject)
 
+  val component = if get?(json, "component") is JObject:
+    ComponentCode(json["component"] as JObject)
+  else:
+    false
+
+  val mounting = if get?(json, "mounting") is String:
+    json["mounting"] as String
+  else:
+    ""
+
+  val vendor-part-numbers = if get?(json, "vendor_part_numbers") is JObject:
+    VendorPartNumbers(json["vendor_part_numbers"] as JObject)
+  else:
+    []
+
```

Comment:
I would slightly prefer it written this way:
```
val component = match(get?(json, "component")) :
  (j:JObject) :
    ComponentCode(j)
  (_) : false

val mounting = match(get?(json, "mounting")) :
  (s:String) : s
  (_) : ""

val vendor-part-numbers = match(get?(json, "vendor_part_numbers")) :
  (j:JObject) : VendorPartNumbers(j)
  (_) : []

Reply:
Thanks - much nicer!